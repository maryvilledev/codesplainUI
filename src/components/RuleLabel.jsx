import React, { PropTypes } from 'react';

const styles = {
  countSpan: {
    color: '#666666',
    display: 'block',
    float: 'right',
  },
  label: {
    borderRadius: '5px',
    borderStyle: 'solid',
    borderWidth: '1px',
    cursor: 'pointer',
    display: 'inline-block',
    fontSize: '10pt',
    lineHeight: 1,
    marginTop: '5px',
    maxWidth: '200px',
    padding: '5px',
    width: '100%',
  },
};

const RuleLabel = ({ color, rule, count, onClick, isActive }) => {
  const backgroundColor = isActive ? color : 'transparent';
  const borderColor = isActive ? 'transparent' : '#e6e6e6' // light grey

  return (
    <div
      style={{...styles.label, backgroundColor, borderColor }}
      onClick={onClick}
    >
      {rule}
      <span style={styles.countSpan}>{`(${count})`}</span>
    </div>
  );
};

RuleLabel.propTypes = {
  color: PropTypes.string.isRequired,
  count: PropTypes.string.isRequired,
  isActive: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  rule: PropTypes.string.isRequired,
};

export default RuleLabel;
