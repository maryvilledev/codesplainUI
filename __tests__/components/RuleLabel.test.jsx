import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import RuleLabel from '../../src/components/RuleLabel';

describe('<RuleLabel />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, { context: { muiTheme } });

  describe('snapshot tests', () => {
    it('matches snapshot of when it is active', () => {
      const wrapper = shallowWithContext(
        <RuleLabel
          color="#FF0000"
          rule="Example Rule"
          count="1"
          isActive={true}
          onClick={jest.fn()}
        />
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('matches snapshot of when it is not active', () => {
      const wrapper = shallowWithContext(
        <RuleLabel
          color="#FF0000"
          rule="Example Rule"
          count="1"
          isActive={false}
          onClick={jest.fn()}
        />
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('prop: rule', () => {
    it('is rendered as a child', () => {
      const rule = 'Those are my eyeholes, man!';
      const wrapper = shallowWithContext(
        <RuleLabel
          color=""
          rule={rule}
          count="0"
          isActive={false}
          onClick={jest.fn()}
        />
      );
      expect(wrapper.childAt(0).text()).toEqual(rule);
    });
  });

  describe('prop: count', () => {
    const count = "3";
    const wrapper = shallowWithContext(
      <RuleLabel
        color=""
        rule=""
        count={count}
        isActive={false}
        onClick={jest.fn()}
      />
    );
    expect(wrapper.find('span').text()).toBe(`(${count})`);
  });

  describe('prop: onClick', () => {
    it('should forward it', () => {
      const onClick = jest.fn();
      const wrapper = shallowWithContext(
        <RuleLabel
          color={''}
          rule={''}
          count={''}
          isActive={false}
          onClick={onClick}
        />
      );
      expect(wrapper.find('div').prop('onClick')).toEqual(onClick);
    });

    it('is called when the component is clicked', () => {
      const onClick = jest.fn();
      const wrapper = shallowWithContext(
        <RuleLabel
          color={''}
          rule={''}
          count={''}
          isActive={false}
          onClick={onClick}
        />
      );
      wrapper.simulate('click');
      expect(onClick).toHaveBeenCalled();
    });
  });
});
