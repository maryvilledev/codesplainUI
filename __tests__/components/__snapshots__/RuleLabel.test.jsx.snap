exports[`<RuleLabel /> snapshot tests matches snapshot of when it is active 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#FF0000",
      "borderColor": "transparent",
      "borderRadius": "5px",
      "borderStyle": "solid",
      "borderWidth": "1px",
      "cursor": "pointer",
      "display": "inline-block",
      "fontSize": "10pt",
      "lineHeight": 1,
      "marginTop": "5px",
      "maxWidth": "200px",
      "padding": "5px",
      "width": "100%",
    }
  }>
  Example Rule
  <span
    style={
      Object {
        "color": "#666666",
        "display": "block",
        "float": "right",
      }
    }>
    (1)
  </span>
</div>
`;

exports[`<RuleLabel /> snapshot tests matches snapshot of when it is not active 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "borderColor": "#e6e6e6",
      "borderRadius": "5px",
      "borderStyle": "solid",
      "borderWidth": "1px",
      "cursor": "pointer",
      "display": "inline-block",
      "fontSize": "10pt",
      "lineHeight": 1,
      "marginTop": "5px",
      "maxWidth": "200px",
      "padding": "5px",
      "width": "100%",
    }
  }>
  Example Rule
  <span
    style={
      Object {
        "color": "#666666",
        "display": "block",
        "float": "right",
      }
    }>
    (1)
  </span>
</div>
`;
